File: 250721/일의 자리 배열/array-with-units.cpp
#include <iostream>
using namespace std;

int main() {
    int first, second;
    cin >> first >> second;
    int arr[10] = {};
    arr[0] = first;
    arr[1] = second;
    for (int i = 2; i < 10; i++) {
        arr[i] = (arr[i - 2] + arr[i - 1]) % 10;
    }
    for (int i = 0; i < 10; i++) {
        cout << arr[i] << " ";
    }
    return 0;
}

File: 250723/일의 자리 배열/array-with-units.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(System.`in`.reader())
    val st = StringTokenizer(br.readLine())
    val (first, second) = Pair(
        st.nextToken().toInt(),
        st.nextToken().toInt()
    )
    br.close()
    val intArray = IntArray(10).apply {
        this[0] = first
        this[1] = second
        indices.drop(2).forEach { i ->
            this[i] = (this[i - 2] + this[i - 1]) % 10
        }
    }
    val result = buildString {
        intArray.forEach {
            append(it)
            append(" ")
        }
    }
    BufferedWriter(System.out.writer()).use {
        it.write(result)
    }
}